gradio/blocks.py

```diff
@@ -657,13 +660,14 @@ def iterate_over_children(children_list):
                         first_dependency = dependency
 
             # Allows some use of Interface-specific methods with loaded Spaces
-            blocks.predict = [fns[0]]
-            blocks.input_components = [
-                Context.root_block.blocks[i] for i in first_dependency["inputs"]
-            ]
-            blocks.output_components = [
-                Context.root_block.blocks[o] for o in first_dependency["outputs"]
-            ]
+            if len(fns):
```

Comment:
Separate issue to be able to load Spaces without any defined event triggers

Reply:
Nice